src/mscorlib/src/System/String.cs

```diff
@@ -159,27 +159,26 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
```

Comment:
That's correct. Mentioned some of the reasoning in the PR message.
There is not a lot in it, for a null separator on List&lt;string&gt;(100) it is 3227ms with it and 3186ms without it. It's just not protecting from anything as `Append` does the check again anyway.